File: AndroidBasics/VideoplayerExa/app/src/main/java/com/mehru/videoplayerexa/MainActivity.java
package com.mehru.videoplayerexa;

import android.media.AudioManager;
import android.media.MediaPlayer;
import android.net.Uri;
import android.os.Bundle;
import android.widget.MediaController;
import android.widget.VideoView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    VideoView videoView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        videoView = findViewById(R.id.videoView);

        String videoPath = "android.resource://"+ getPackageName()+"/raw/clip_1";


        Uri vURI = Uri.parse(videoPath);
        videoView.setVideoURI(vURI);
        videoView.start();
                 //add options
        MediaController mediaController = new MediaController(this);
        videoView.setMediaController(mediaController);
        mediaController.setAnchorView(videoView);
    }
}

File: app dev kt/RecyclerView/app/src/main/java/com/mehru/recyclerview/MainActivity.kt
package com.mehru.recyclerview

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView : RecyclerView
    private lateinit var newsArrayList : ArrayList<News>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recyclerView)

        val newsImageArray = arrayOf(
            R.drawable.contac_logo10,
            R.drawable.contac_logo13,
            R.drawable.contac_logo11,
            R.drawable.contac_logo12,
            R.drawable.contac_logo13,
            R.drawable.contac_logo14,
            R.drawable.contac_logo13,
            R.drawable.contac_logo11,
            R.drawable.contac_logo12,
            R.drawable.contac_logo13
        )

        val newsHeadingArray = arrayOf(
            "Government Launches New Digital Education Program for Rural Students",
            "India and Japan Sign Strategic Deal on Renewable Energy Projects" ,
            "Tech Giants Announce 20,000+ Job Openings for Fresh Graduates",
            "India Wins Thriller Against Australia in ICC T20 World Cup Final",
            "Heavy Rainfall Alert Issued in Northern India for Next 3 Days",
            "Bollywood Star Announces Comeback with Big-Budget Sci-Fi Film",
            "India and Japan Sign Strategic Deal on Renewable Energy Projects" ,
            "Tech Giants Announce 20,000+ Job Openings for Fresh Graduates",
            "India Wins Thriller Against Australia in ICC T20 World Cup Final",
            "Heavy Rainfall Alert Issued in Northern India for Next 3 Days"
        )

        recyclerView.layoutManager = LinearLayoutManager(this)
        newsArrayList = arrayListOf<News>()

        for (index in newsImageArray.indices){
            val news = News(newsHeadingArray[index],newsImageArray[index])
            newsArrayList.add(news)
        }

        recyclerView.adapter = NewsAdapter(newsArrayList,this)

        
    }
}